### PR TITLE
promql/offset check

### PR DESCRIPTION
**Pull request by Tarmander**
Adds promql/offset check

### CODE REVIEW

File: internal/checks/promql_offset.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	for _, selector := range getSelectorsWithOffset(expr.Query) {
+		match, _ := regexp.MatchString(c.prefixRegex.anchored, selector.Name)
+		if match && c.min > selector.OriginalOffset {
+			problems = append(problems, Problem{
+				Fragment: selector.String(),
+				Lines:    expr.Lines(),
+				Reporter: c.Reporter(),
+				Text:     fmt.Sprintf("the %s metric requires a minimum offset of %s", selector.String(), c.min.String()),
+				Severity: Warning,
```
Comment (thejuan) [inline internal/checks/promql_offset.go]: I think we should allow the Severity to be dynamic via config (but not required, default to wanring)
Why did you go with Warning? For our use case, if it fails isn't it an error? Or are there edge cases?

File: docs/checks/promql/offset.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+to work.
+To enable it add one or more `rule {...}` blocks and specify all required
+rules there.
+
+Examples:
+
+Ensure that all metrics beginning with `aws_` have a minimum offset of 5 minutes:
+
+```js
+rule {
```
Comment (thejuan) [inline docs/checks/promql/offset.md:41]: Maybe add an example for multiple offsets

**Review by thejuan — COMMENTED** [review]
Some comments, but looks great!

File: internal/checks/promql_offset.go (hunk 1 above)
Comment (caryjpaepen) [reply internal/checks/promql_offset.go]: Maybe Kevin's worried about `aws_.*` metrics that are not from Cloudwatch. I don't know if there is any in our prod environment, worth checking. 

The alternative might be to check for specific label values (in our case, we have something like `config="yace"` or `config='cloudwatch-..."`), but that requires querying every single prom node again, so probably too much work/querying for little gain.

File: internal/checks/promql_offset_test.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	"github.com/cloudflare/pint/internal/checks"
+	"github.com/cloudflare/pint/internal/promapi"
+)
+
+func newOffsetCheck(_ *promapi.FailoverGroup) checks.RuleChecker {
+	dur, _ := time.ParseDuration("5m")
+	return checks.NewOffsetCheck(checks.MustTemplatedRegexp("aws_.*"), dur)
+}
+
+func TestOffsetCheck(t *testing.T) {
```
Comment (caryjpaepen) [inline internal/checks/promql_offset_test.go:16]: Should we add recording rule test(s) just to be sure that's also working?

**Comment by caryjpaepen** [comment]
Algo looks great

File: internal/checks/promql_offset_test.go (hunk 3 above)
Comment (Tarmander) [reply internal/checks/promql_offset_test.go:16]: Good idea

File: docs/checks/promql/offset.md (hunk 2 above)
Comment (Tarmander) [reply docs/checks/promql/offset.md:41]: Good catch. I'll add those in.

File: internal/checks/promql_offset.go (hunk 1 above)
Comment (Tarmander) [reply internal/checks/promql_offset.go]: This was my fear: that we'd be enforcing an offset with a regex match, but there is nothing in place to prohibit a dev from creating an `aws_.*` metric in the future that doesn't require an offset. 

I like the idea of querying for matching label values... but is that worth the extra querying time on Pint just to enforce an offset? I think the regex match will cover all cloud metrics, and in the above niche case we can override the validation.

Comment (thejuan) [reply internal/checks/promql_offset.go]: Agree the label checks aren't for now

**Review by thejuan — APPROVED** [review]

**Pull request merged**